src/Engine/ProtoScript/Runners/LiveRunner.cs

```diff
@@ -1028,6 +1031,52 @@ public interface ILiveRunner
 
     public partial class LiveRunner : ILiveRunner, IDisposable
     {
+        internal class DebugByteCodeMode : IDisposable
+        {
+            private TextWriter oldAsmOutput;
+            private bool oldDumpByteCode;
+            private bool oldVerbose;
+            private ExecutionMode oldExecutionMode;
+
+            private Core core;
+
+            internal DebugByteCodeMode(Core core)
+            {
+                // Setup the log file location
+                var logFolder = Path.Combine(Directory.GetCurrentDirectory(), "ByteCodeLogs");
+                Directory.CreateDirectory(logFolder);
+                var newWriter = File.CreateText(Path.Combine(logFolder, DateTime.Now.ToString("yyyy-MM-ddTHH-mm-ss-fff") + ".log"));
```

Comment:
should this be wrapped in a try catch?

Reply:
so to enable or disable logging you need to turn this flag on(or off) and then reboot Dynamo?

Reply:
no....you can start dynamo and enable the debug mode to start generating logs.
no need for a reboot.

Not sure a try catch is worth it...this is going to be used by devs to debug

Reply:
ok so `SynchronizeInternal` gets called a lot - I guess this means we might have log files per each run - that seems good. 👍